### PR TITLE
Add guide for `@relayId`

### PR DESCRIPTION
**Pull request by darrellwarde**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
+[[relay-compatibility]]
+= Relay Compatibility
+
+The https://relay.dev/docs/guides/graphql-server-specification/[Relay GraphQL Server Specification] states that in order for a server to be compatibile with Relay, it must provide:
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
The https://relay.dev/docs/guides/graphql-server-specification/[Relay GraphQL Server Specification] states that in order for a server to be compatible with Relay, it must provide:
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+1. A mechanism for refetching an object.
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
. A mechanism for refetching an object.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+2. A description of how to page through connections.
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
. A description of how to page through connections.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+The Neo4j GraphQL Library provides a description of how to page through connections out of the box, however, configuration is required for it to provide a mechanism for refetching an object, which this guide details.
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
The Neo4j GraphQL Library provides a description of how to page through connections out of the box.
However, configuration is required for it to provide a mechanism for refetching an object, which this page details.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 5]
```diff
@@ -0,0 +1,76 @@
+[[relay-compatibility]]
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc:1]:
```suggestion
[[relay-compatibility]]
:description: This page describes how to set up a mechanism for refetching an object when using a Relay GraphQL server.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+= Relay Compatibility
+
+The https://relay.dev/docs/guides/graphql-server-specification/[Relay GraphQL Server Specification] states that in order for a server to be compatibile with Relay, it must provide:
+
+1. A mechanism for refetching an object.
+2. A description of how to page through connections.
+
+The Neo4j GraphQL Library provides a description of how to page through connections out of the box, however, configuration is required for it to provide a mechanism for refetching an object, which this guide details.
+
+== Object Identification
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
== Object identification
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+In order for a Relay client to be able to refetch objects, each type must have a unique object indentifier which can be used for this purpose.
+
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+A server informs a Relay client that this requirement has been satisfied by having types implement the `Node` directive, which has the following definition:
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
A server then informs a Relay client that this requirement has been satisfied by having types that implement the `Node` directive, which has the following definition:
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+[source, graphql]
+----
+type Book implements Node {
+  id: ID!
+  isbn: String!
+}
+----
+
+With the above definition, Relay clients will be able to refetch `Book` objects using the `node` query field.
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
With that, Relay clients are able to refetch `Book` objects using the `node` query field.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+With the above in mind, the following section dives into how to do this with the Neo4j GraphQL Library.
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
The following section dives into how to do this with the Neo4j GraphQL Library.
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+The Neo4j GraphQL Library provides an abstraction over the required `id` field, allowing the value of any unique property in the database to be used as the object identifier.
+
+This is provided by the `@relayId` directive, which has the following definition:
+
+[source, graphql]
+----
+directive @relayId on FIELD_DEFINITION
+----
+
+Using the `Book` example above, ISBN is normally a unique identifier. So you configure this like so:
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
Considering the same `Book` example, ISBN is normally a unique identifier. 
You can configure this like so:
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+[source, graphql]
+----
+type Book implements Node {
+  id: ID!
+  isbn: String!
+}
+----
+
+When the `id` field is used, `isbn` is the underlying property that is used in the database.
+
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
```

File: modules/ROOT/pages/integrations/relay-compatibility.adoc [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+The `Book` type is now refetchable via the `node` query field, and is ready to be used with a Relay client!
```
Comment (lidiazuin) [inline modules/ROOT/pages/integrations/relay-compatibility.adoc]:
```suggestion
Also, the `Book` type is now refetchable via the `node` query field, and is ready to be used with a Relay client.
```

**Review by lidiazuin — APPROVED** [review]
Some editorial suggestions! Thank you.

**Review by mjfwebb — APPROVED** [review]

**Pull request merged**